BrainPortal/app/controllers/groups_controller.rb

```diff
@@ -248,7 +248,7 @@ def switch #:nodoc:
     elsif params[:id] == "all"
       cbrain_session[:active_group_id] = "all"
     else
-      @group = current_user.public_and_available_groups.find(params[:id])
+      @group = current_user.viewable_groups.find(params[:id])
```

Comment:
I am not able to switch to a public project as a SiteManager, when I am on the show page of a public group. I do not have the button to do that. But I can switch with the switch link in the index page of the group (list view).